lib/numerizer.rb

```diff
@@ -108,50 +105,105 @@ class Numerizer
+  ALL_FRACTIONS = FRACTIONS + (SINGLE_ORDINAL_FRACTIONALS + DIRECT_ORDINAL_FRACTIONALS).map {|tp| [tp[0] + '(s)?', tp[1]] }
+  ONLY_PLURAL_FRACTIONS = FRACTIONS + (SINGLE_ORDINAL_FRACTIONALS + DIRECT_ORDINAL_FRACTIONALS).map {|tp| [tp[0] + 's', tp[1]] }
+
+  ALL_ORDINALS_REGEX = ALL_ORDINALS.reduce { |a,b| a[0] + '|' + b[0] }
+
+
+  def self.numerize(string, ignore: [], bias: :none)
     string = string.dup
+    ignore = ignore.map(&:downcase).to_set
 
     # preprocess
     string.gsub!(/ +|([^\d])-([^\d])/, '\1 \2') # will mutilate hyphenated-words
     string.gsub!(/\ba$/, '') && string.rstrip! # doesn't make sense for an 'a' at the end to be a 1
 
     # easy/direct replacements
     (DIRECT_NUMS + SINGLE_NUMS).each do |dn|
-      string.gsub!(/(^|\W)#{dn[0]}(?=$|\W)/i, '\1<num>' + dn[1].to_s)
+      next if ignore.include? dn[0].downcase 
+      string.gsub!(/(^|\W)#{dn[0]}(?=$|\W)/i, '\1<num>' + dn[1].to_s) 
     end
 
     # ten, twenty, etc.
     TEN_PREFIXES.each do |tp|
+      next if ignore.include? tp[0].downcase 
       SINGLE_NUMS.each do |dn|
+        next if ignore.include? dn[0].downcase 
         string.gsub!(/(^|\W)#{tp[0]}#{dn[0]}(?=$|\W)/i, '\1<num>' + (tp[1] + dn[1]).to_s)
       end
-      SINGLE_ORDINALS.each do |dn|
+      (ORDINALS + SINGLE_ORDINAL_FRACTIONALS).each do |dn|
+        next if ignore.include? dn[0].downcase 
         string.gsub!(/(^|\W)#{tp[0]}(\s)?#{dn[0]}(?=$|\W)/i, '\1<num>' + (tp[1] + dn[1]).to_s + dn[0][-2, 2])
       end
       string.gsub!(/(^|\W)#{tp[0]}(?=$|\W)/i, '\1<num>' + tp[1].to_s)
     end
 
     # handle fractions
-    FRACTIONS.each do |tp|
+    # only plural fractions if ordinal mode
+    if bias == :ordinal
+      fractionals = ONLY_PLURAL_FRACTIONS 
+    else
+      fractionals = ALL_FRACTIONS
+    end
+
+    fractionals.each do |tp|
+      next unless ignore.select {|x| /#{tp[0]}/ =~ x } .empty?
       string.gsub!(/a #{tp[0]}(?=$|\W)/i, '<num>1/' + tp[1].to_s)
-      string.gsub!(/\s#{tp[0]}(?=$|\W)/i, '/' + tp[1].to_s)
+      # TODO : Find Noun Distinction for Quarter
+      # Handle Edge Case with Quarter
+      if bias == :fractional
+        string.gsub!(/(^|\W)#{tp[0]}(?=$|\W)/i, '/' + tp[1].to_s)
+      elsif /quarter/ =~ tp[0] then
+        string.gsub!(/(?:\w*)(^|\W)#{tp[0]}(?=$|\W)/i) do |match|
+          if (match =~ /^(i|you|he|she|we|it|you|they|to|the)/i) == nil then 
+            match.gsub!(/(^|\W)#{tp[0]}/, '/' + tp[1].to_s)
+          end
+          match
+        end
+      else
+        string.gsub!(/(?<!the)\s#{tp[0]}(?=$|\W)/i, '/' + tp[1].to_s)
+      end
     end
 
-    (DIRECT_ORDINALS + SINGLE_ORDINALS).each do |on|
-      string.gsub!(/(^|\W)#{on[0]}(?=$|\W)/i, '\1<num>' + on[1].to_s + on[0][-2, 2])
+    ALL_ORDINALS.each do |on|
+      break if bias == :fractional # shouldn't be necessary but saves cycles
+      next if ignore.include? on[0].downcase 
+      if on[0] == 'second' and bias != :ordinal then
```

Comment:
I prefer `&&` instead of `and`. `and` has a lower precedence which can cause some odd behaviour and should usually be avoided except in very special circumstances. More [here](https://stackoverflow.com/questions/1426826/difference-between-and-and-in-ruby) if you're interested.